src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -412,8 +416,18 @@ - (void)_handleEventReport:(NSArray<NSDictionary<NSString *, id> *> *)eventRepor
         MTREventPath * eventPath = eventDict[MTREventPathKey];
         BOOL isStartUpEvent = (eventPath.cluster.unsignedLongValue == MTRClusterIDTypeBasicInformationID)
             && (eventPath.event.unsignedLongValue == MTREventIDTypeClusterBasicInformationEventStartUpID);
-        if (isStartUpEvent) {
-            _estimatedStartTime = nil;
+        // StartUp event would be received when server resumes subscription
+        if (isStartUpEvent && (_state == MTRDeviceStateReachable)) {
```

Comment:
So this depends on us _not_ setting state to reachable during priming.

I would really much rather we had an isPriming helper or something that we could use to detect "we are in priming", and then we implement that in terms of the state and whatnot.  It would make it easier to fix things up if we change our state transitions at some point.

Followup is fine for adding this helper.